File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rsp/keys/api/KeysImpl.java
package org.ngbp.jsonrpc4jtestharness.rsp.keys.api;

import org.ngbp.jsonrpc4jtestharness.models.JsonRpcResponse;
import org.ngbp.jsonrpc4jtestharness.rsp.filterCodes.model.GetFilterCodes;

public class KeysImpl implements Keys {
    @Override
    public JsonRpcResponse<GetFilterCodes> requestKeys() {
        return null;
    }

    @Override
    public JsonRpcResponse<Object> relinquishKeys() {
        return null;
    }

    @Override
    public JsonRpcResponse<Object> requestKeysTimeout() {
        return null;
    }
}

File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/keys/KeysImpl.kt
package org.ngbp.jsonrpc4jtestharness.rpc.keys

import org.ngbp.jsonrpc4jtestharness.rpc.RpcResponse
import org.ngbp.jsonrpc4jtestharness.rpc.keys.model.Keys

class KeysImpl() : IKeys {
    override fun requestKeys(listOfKeys: List<String>): Keys {
        return Keys().apply {
            this.accepted = listOfKeys
        }
    }

    override fun relinquishKeys(listOfKeys: List<String>): RpcResponse {
        return RpcResponse()
    }
}